### PR TITLE
RELATED: RAIL-2019 Document text filter character escaping

### PR DESCRIPTION
**Pull request by Tulean**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/232/ : SUCCESS in 1m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/87/ : SUCCESS in 9s

File: docs/afm.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ const afm = {
 Attribute filters (both `positiveAttributeFilter` and `negativeAttributeFilter`) can be defined to match attribute elements by their URI (this is the default) or value (text filter). 
 To use text filters, define the `textFilter` property of the filter and use values instead of URIs in the `in` or `notIn` arrays. 
 
+In scenarios where escaping characters are needed while using text filters, please make sure to check whether input is correctly escaped. Inputs that are not correctly escaped could potentially throw errors. Please note that single quotes and double quotes behave differently while escaping characters.
```
Comment (no23reason) [inline docs/afm.md]: We should list the characters that can cause problems. Also, some examples would be useful

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/234/ : SUCCESS in 1m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/90/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/236/ : SUCCESS in 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/91/ : SUCCESS in 5s

File: docs/afm.md [hunk 2]
```diff
@@ -63,6 +63,10 @@ const afm = {
 Attribute filters (both `positiveAttributeFilter` and `negativeAttributeFilter`) can be defined to match attribute elements by their URI (this is the default) or value (text filter). 
 To use text filters, define the `textFilter` property of the filter and use values instead of URIs in the `in` or `notIn` arrays. 
 
+In scenarios where escaping characters are needed while using text filters, please make sure to check whether input is correctly escaped. Inputs that are not correctly escaped could potentially throw errors. 
+
+Characters that are prone to mistakes are `"` and `\`. For example, if the user wants to filter out this exact string `Foo\"`, the input needs to be in this format `'Foo\\"'` or `"Foo\\\""`, both inputs will result in `"Foo\\\""`. The browser will send `"Foo\\\""` to the server where `"Foo\\\""` will be queried as `Foo\"`. Please note that single quotes and double quotes behave differently while escaping characters.
+
```
Comment (evgeniabogranskaya) [inline docs/afm.md]: If you need to escape characters while using text filters, make sure that the input for a filter is escaped correctly. Incorrectly escaped input may cause errors.

Characters that are prone to mistakes are `"` and `\`. For example, if you want to filter out this exact string: `Foo\"`, enter it as `'Foo\\"'` or `"Foo\\\""`, which will result in `"Foo\\\""`. The browser will send `"Foo\\\""` to the server where `"Foo\\\""` will be queried as `Foo\"`.
**NOTE:** Single quotes and double quotes behave differently while escaping characters.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/238/ : SUCCESS in 1m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/93/ : SUCCESS in 4s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/239/ : SUCCESS in 49s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/94/ : SUCCESS in 4s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/240/ : SUCCESS in 42s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/95/ : SUCCESS in 4s

**Comment by evgeniabogranskaya** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/241/ : SUCCESS in 1m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-merge-compliance-check/72/ : SUCCESS in 21s

**Pull request merged**